packages/app/src/hooks/useChartConfig.tsx

```diff
@@ -114,11 +118,22 @@ export function useQueriedChartConfig(
           }
 
           const timestampColumn = inferTimestampColumn(resultSet.meta ?? []);
+          const numericColumn = inferNumericColumn(resultSet.meta ?? []);
+          const numericColumnName = numericColumn?.[0]?.name;
           for (const row of resultSet.data) {
+            const _rowWithoutValue = numericColumnName
+              ? Object.fromEntries(
+                  Object.entries(row).filter(
+                    ([key]) => key !== numericColumnName,
+                  ),
+                )
+              : { ...row };
             const ts =
               timestampColumn != null
                 ? row[timestampColumn.name]
-                : '__FIXED_TIMESTAMP__';
+                : isTimeSeries
+                  ? objectHash(_rowWithoutValue)
```

Comment:
if its a time-series chart but no timestamp is selected, we want to group by the rest of keys